File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/models2.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import OpenAiModel, OpenAiModelDeleteResponse, OpenAiModelList

class Models2Api:
    """Models API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def retrieve_model(self, model: str) -> OpenAiModel:
        """Retrieve model"""
        return self._client.get(f"/ai/v3/v1/models/{model}")

    def delete_model(self, model: str) -> OpenAiModelDeleteResponse:
        """Delete model"""
        return self._client.delete(f"/ai/v3/v1/models/{model}")

    def list_models(self) -> OpenAiModelList:
        """List models"""
        return self._client.get(f"/ai/v3/v1/models")

File: sdkwork-ai-sdk-flutter/lib/src/api/models2.dart
import '../http/client.dart';
import '../models.dart';

class Models2Api {
  final HttpClient _client;
  
  Models2Api(this._client);

  /// Retrieve model
  Future<OpenAiModel?> retrieveModel(String model) async {
    final response = await _client.get(ApiPaths.aiPath('/v1/models/${model}'));
    return response is OpenAiModel ? response : null;
  }

  /// Delete model
  Future<OpenAiModelDeleteResponse?> deleteModel(String model) async {
    final response = await _client.delete(ApiPaths.aiPath('/v1/models/${model}'));
    return response is OpenAiModelDeleteResponse ? response : null;
  }

  /// List models
  Future<OpenAiModelList?> listModels() async {
    final response = await _client.get(ApiPaths.aiPath('/v1/models'));
    return response is OpenAiModelList ? response : null;
  }
}
